src/test/groovy/edu/ie3/util/StringUtilsTest.groovy

```diff
@@ -211,4 +213,109 @@ class StringUtilsTest extends Specification {
+		] as LinkedHashMap<String, String>
+
+		when:
+		def actualList = input.entrySet().stream().map({ mapEntry ->
+			return new AbstractMap.SimpleEntry<String, String>(StringUtils.csvString(mapEntry.key, ","), StringUtils.csvString(mapEntry.value, ","))
+		}) as Set
+
+		def actual = actualList.collectEntries {
+			[it.key, it.value]
+		}
+
+		then:
+		actual == expected
+	}
+
+	def "The StringUtils converts a given String to match the csv specification RFC 4180 "() {
+		expect:
+		StringUtils.csvString(inputString, csvSep) == expect
+
+		where:
+		inputString                                                                         | csvSep || expect
+		"activePowerGradient"                                                               | ","    || "activePowerGradient"
+		"\"100,0\""                                                                         | ","    || "\"100,0\""
+		"100,0"                                                                             | ","    || "\"100,0\""
+		"100,0"                                                                             | ";"    || "\"100,0\""
+		"100;0"                                                                             | ";"    || "\"100;0\""
+		"\"100;0\""                                                                         | ";"    || "\"100;0\""
+		"100;0"                                                                             | ","    || "100;0"
+		"olm:{(0.00,1.00)}"                                                                 | ","    || "\"olm:{(0.00,1.00)}\""
+		"olm:{(0.00,1.00)}"                                                                 | ";"    || "\"olm:{(0.00,1.00)}\""
+		"{\"type\":\"Point\",\"coordinates\":[7.411111,51.492528]}"                         | ","    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"{\"type\":\"Point\",\"coordinates\":[7.411111,51.492528]}"                         | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""         | ","    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""         | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\"\":\"\"Point\"\"\"\",\"\"coordinates\"\"\":[7.411111,51.492528]}\"" | ","    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"\"{\"\"type\"\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""       | ";"    || "\"{\"\"type\"\":\"\"Point\"\",\"\"coordinates\"\":[7.411111,51.492528]}\""
+		"uu,id"                                                                             | ","    || "\"uu,id\""
+		"uu,id"                                                                             | ";"    || "\"uu,id\""
```

Comment:
Please add `"This" is a test`, `This is "a" test` and `This is a "test"` as test cases to make sure quoted sub Strings are handled correctly.